### PR TITLE
site: announce yuque welfare program

### PR DESCRIPTION
**Pull request by chen201724**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### 🚀 Summary
Add a promotional banner for the Yuque welfare program at the top of the website.

### <samp>🤖 Generated by Copilot at d2e3326</samp>

Added a banner to the header to announce the antd 4.0 release and adjusted the nav-phone-icon position to match. The banner uses the `Alert` component and can be closed by the user. The changes affect the files `.dumi/theme/slots/Header/index.tsx` and `.dumi/theme/common/styles/Responsive.tsx`.

### <samp>🤖 Generated by Copilot at d2e3326</samp>

*  Add a banner component to the header component to display a message about the antd 4.17.0 release ([link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728L7-R7), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R23-R39), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R126-R138), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R144), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728L120-R159), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R182-R187), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728L212-R251), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R364-R385))
*  Adjust the position of the nav-phone-icon to align with the banner component ([link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-86c582e67d9d7b7e12b132dd7b7750743e1849130e18187bcf7fbc84f3f606b3L13-R13))
*  Use the locales object and the localStorage item to determine the banner message and visibility based on the user's language and preference ([link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R23-R39), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728L120-R159), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R182-R187))
*  Use the useStyle hook and the banner and link styles to customize the appearance of the banner component ([link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R126-R138), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R364-R385))
*  Use the onBannerVisibleChange callback and the bannerVisible property to handle the onClose event of the banner component and update the header state and the localStorage item accordingly ([link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R144), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R182-R187), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728L212-R251), [link](https://github.com/ant-design/ant-design/pull/44884/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728R364-R385))

### CODE REVIEW

File: .dumi/theme/common/styles/Responsive.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ export default () => {
       styles={css`
         .nav-phone-icon {
           position: absolute;
-          top: 25px;
+          bottom: 17px;
```
Comment (chen201724) [inline .dumi/theme/common/styles/Responsive.tsx:13]: 这里原来根据 top 进行定位会导致 Header 高度被 banner 撑开后定位偏移，所以修改为通过 bottom 定位，避免对应的高度影响

File: .dumi/theme/slots/Header/index.tsx [hunk 2]
```diff
@@ -20,6 +20,23 @@ import type { SharedProps } from './interface';
 const RESPONSIVE_XS = 1120;
 const RESPONSIVE_SM = 1200;
 
+const locales = {
```
Comment (zombieJ) [inline .dumi/theme/slots/Header/index.tsx:23]: 写死不太好，换成 h5 data （感觉会影响 CLS）。 
@MadCcc

File: .dumi/theme/slots/Header/index.tsx [hunk 3]
```diff
@@ -20,6 +20,23 @@ import type { SharedProps } from './interface';
...
+      '语雀公益计划：大学生认证教育邮箱，即可免费获得语雀会员。语雀，支付宝匠心打造的在线文档平台。',
+    shortMessage: '支付宝语雀·大学生公益计划火热进行中！',
+    more: '了解更多',
+  },
+  en: {
+    message:
+      'By verifying education Email, college students can get free member of Yuque. Yuque is an online document tool created by Alipay.',
+    shortMessage: "Yuque's welfare activity is in progress! ",
+    more: 'Learn more',
+  },
```
Comment (afc163) [inline .dumi/theme/slots/Header/index.tsx:35]: 英文页面是不是不需要展示？这个推广更多是面向国内的

File: .dumi/theme/slots/Header/index.tsx [hunk 4]
```diff
@@ -20,6 +20,23 @@ import type { SharedProps } from './interface';
 const RESPONSIVE_XS = 1120;
 const RESPONSIVE_SM = 1200;
 
+const locales = {
+  cn: {
+    message:
+      '语雀公益计划：大学生认证教育邮箱，即可免费获得语雀会员。语雀，支付宝匠心打造的在线文档平台。',
+    shortMessage: '支付宝语雀·大学生公益计划火热进行中！',
```
Comment (afc163) [inline .dumi/theme/slots/Header/index.tsx:27]:
```suggestion
    shortMessage: '支付宝语雀 · 大学生公益计划火热进行中！',
```

File: .dumi/theme/slots/Header/index.tsx (hunk 2 above)
Comment (chen201724) [reply .dumi/theme/slots/Header/index.tsx:23]: > 写死不太好，换成 h5 data （感觉会影响 CLS）。 @MadCcc

没太理解是什么意思，可以提供一个简单的代码示例吗？

File: .dumi/theme/slots/Header/index.tsx [hunk 5]
```diff
@@ -322,6 +361,29 @@ const Header: React.FC = () => {
...
+                className={styles.link}
+                href="https://www.yuque.com/yuque/blog/welfare-edu?source=antd"
+                target="_blank"
+                rel="noreferrer"
+              >
+                {locale.more}
+              </a>
+            </>
+          }
+          type="success"
```
Comment (afc163) [inline .dumi/theme/slots/Header/index.tsx:380]:
```suggestion
          type="info"
```

File: .dumi/theme/slots/Header/index.tsx (hunk 3 above)
Comment (xudafeng) [reply .dumi/theme/slots/Header/index.tsx:35]: 就纯中文吧。

@chen201724

File: .dumi/theme/slots/Header/index.tsx [hunk 6]
```diff
@@ -322,6 +361,29 @@ const Header: React.FC = () => {
...
+        <Alert
+          className={styles.banner}
+          message={
+            <>
+              {isMobile ? locale.shortMessage : locale.message}
+              <a
+                className={styles.link}
+                href="https://www.yuque.com/yuque/blog/welfare-edu?source=antd"
+                target="_blank"
+                rel="noreferrer"
```
Comment (afc163) [inline .dumi/theme/slots/Header/index.tsx:374]:
```suggestion
                rel="noreferrer"
                onClick={() => {
                  window.gtag?.('event', '点击', {
                    event_category: 'top_banner',
                    event_label: https://www.yuque.com/yuque/blog/welfare-edu?source=antd,
                  });
                }}
```

File: .dumi/theme/slots/Header/index.tsx (hunk 3 above)
Comment (chen201724) [reply .dumi/theme/slots/Header/index.tsx:35]: ok

File: .dumi/theme/slots/Header/index.tsx [hunk 7]
```diff
@@ -322,6 +361,29 @@ const Header: React.FC = () => {
           <MenuOutlined className="nav-phone-icon" onClick={handleShowMenu} />
         </Popover>
       )}
+      {bannerVisible && (
```
Comment (MadCcc) [inline .dumi/theme/slots/Header/index.tsx:364]: 这个会影响整个页面的 CLS，最好还是做成不可关闭或者 fixed 的

**Review by MadCcc — APPROVED** [review]

**Pull request merged**